ekncontent/ekncontent/eknc-domain.c

```diff
@@ -0,0 +1,862 @@
+  return g_file_get_child (dir, self->subscription_id);
+}
+
+static GFile *
+eknc_domain_get_manifest_file (EkncDomain *self)
+{
+  g_autoptr(GFile) dir = eknc_domain_get_subscriptions_dir (self);
+  return g_file_get_child (dir, "manifest.json");
+}
+
+static gboolean
+eknc_domain_clean_old_symlinks (EkncDomain *self,
+                                GCancellable *cancellable,
+                                GError **error)
+{
+  GError *local_error = NULL;
+  g_autoptr(GFile) subscription_dir = eknc_domain_get_subscriptions_dir (self);
+  g_autoptr(GFileEnumerator) enumerator = g_file_enumerate_children (subscription_dir,
+                                                                     "standard::name,standard::type",
+                                                                     G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
+                                                                     cancellable,
+                                                                     &local_error);
+  if (local_error != NULL)
+    {
+      g_propagate_error (error, local_error);
+      return FALSE;
+    }
+
+  gpointer info_ptr;
+  while ((info_ptr = g_file_enumerator_next_file (enumerator, cancellable, &local_error)))
+    {
+      g_autoptr(GFileInfo) info = info_ptr;
+      g_autoptr(GFile) file = g_file_enumerator_get_child (enumerator, info);
+      if (!g_file_query_exists (file, cancellable) && !g_file_delete (file, cancellable, &local_error))
+        {
+          g_propagate_error (error, local_error);
+          return FALSE;
+        }
+    }
+
+  if (local_error != NULL)
+    {
+      g_propagate_error (error, local_error);
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static gboolean
+eknc_domain_setup_link_tables (EkncDomain *self,
+                               GError **error)
+{
+  for (GSList *l = self->shards; l; l = l->next)
+    {
+      EosShardShardFile *shard = l->data;
+      EosShardRecord *record = eos_shard_shard_file_find_record_by_hex_name (shard, LINK_TABLE_ID);
+      if (record == NULL)
+        continue;
+      EosShardDictionary *dict;
+      if (!(dict = eos_shard_blob_load_as_dictionary (record->data, error)))
+        return FALSE;
+      self->link_tables = g_slist_append (self->link_tables, dict);
+    }
+
+  return TRUE;
+}
+
+static gboolean
+eknc_domain_initable_init (GInitable *initable,
+                           GCancellable *cancellable,
+                           GError **error)
+{
+  EkncDomain *self = EKNC_DOMAIN (initable);
+
+  if (self->app_id == NULL || *self->app_id == '\0')
+    g_set_error (error, EKNC_DOMAIN_ERROR, EKNC_DOMAIN_ERROR_APP_ID_NOT_SET,
+                 "You must set an app id to initialize a domain object");
+
+  self->content_dir = eknc_get_data_dir (self->app_id);
+  if (!eknc_load_subscription_entries (self, cancellable, error))
+    return FALSE;
+
+  g_autoptr(GFile) manifest_file = eknc_domain_get_manifest_file (self);
```

Comment:
Since the first two of these are used in multiple places, and must be fetched here in `initable_init` in any case, consider sticking them in `priv` and only fetching/unreffing them once.